### ISSUE TITLE
[HOLD for payment 2023-08-07] [$250] Migrate InitialSettingsPage.js to function component

### ISSUE THREAD

**Issue by marcaaron** [Expensify/App#16281]
### Filenames
- [ ] [src/pages/settings/InitialSettingsPage.js](https://github.com/Expensify/App/blob/main/src/pages/settings/InitialSettingsPage.js)
    - **type:** `React.Component`
    - **has state values:** true
    - **has refs:** false
    - **has context:** false
    - **lifecycle methods:** `componentDidMount`

### Task
- We currently have some class components in our codebase that we would like to refactor to a function component.
- Here's a link with some general advice on how to refactor a class component to a function component: https://react.dev/reference/react/Component#alternatives
- If you need additional guidance, please ask in [#expensify-open-source](https://expensify.slack.com/archives/C01GTK53T8Q)
- Test for any regressions and verify that there are no breaking changes

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01ed00725204fbc329</li>
        <li>Upwork Job ID: 1673399218372874240</li>
        <li>Last Price Increase: 2023-06-26</li>
    </ul>
</details>

**Comment by MelvinBot** [comment]
⚠️ Could not update price automatically because there is no linked Upwork Job ID. The BZ team member will need to update the price manually in Upwork.

**Comment by marcaaron** [comment]
Heads up! The pricing for this issue has been adjusted based on the scope of the work and the fact that no complex bug investigations or proposal are required.

**Comment by muxriddinmuqimov77** [comment]
I'd like to work on this issue when it goes to external

**Comment by neonbhai** [comment]
I would love to work on this!

**Comment by sophiepintoraetz** [comment]
Good catch, @0xmiroslav - @neonbhai is next on the list so shall we go with them?

**Comment by yh-0218** [comment]
I can work on this

**Comment by vdem0115** [comment]
I'd like to work on this issue.

**Comment by vdem0115** [comment]
Contributor details
Your Expensify account email: volodymyrdemydenko07@gmail.com
Upwork Profile Link: https://www.upwork.com/freelancers/~01e385620a0a8f347c

**Comment by code0406** [comment]
I would like to work on this task.

**Comment by yh-0218** [comment]
I can do this in a day.

**Comment by vdem0115** [comment]
Hello
I would like to work on this issue

**Comment by sophiepintoraetz** [comment]
Just waiting on @thesahindia to review the PR!

**Comment by yh-0218** [comment]
@thesahindia PR is ready for your review. Thanks.

**Comment by sophiepintoraetz** [comment]
Bump @thesahindia!

**Comment by yh-0218** [comment]
Hi, @sophiepintoraetz @rushatgabhane I just fixed conflict. please check [PR](https://github.com/Expensify/App/pull/22153). Thanks.

**Comment by yh-0218** [comment]
Hi,  @sophiepintoraetz @rushatgabhane This task assigned me. right?
https://github.com/Expensify/App/issues/16281#issuecomment-1612296625

**Comment by sophiepintoraetz** [comment]
Yes, this task is assigned to you! Just waiting on @rushatgabhane to review the PR.

**Comment by rushatgabhane** [comment]
thanks!

**Comment by yh-0218** [comment]
Hi, @thesahindia @rushatgabhane Could you check this PR. Thanks.

**Comment by sophiepintoraetz** [comment]
@rushatgabhane - please can we get this PR reviewed? It's been a couple of weeks now!

**Comment by iwiznia** [comment]
Going on sabbatical, please reassign another engineer

**Comment by sophiepintoraetz** [comment]
Bumped in slack

**Comment by sophiepintoraetz** [comment]
Just assigning another engineer for the review - Daniel! (Ioni is on sabbatical)

**Comment by rushatgabhane** [comment]
sorry, i can't review this https://expensify.slack.com/archives/C02NK2DQWUX/p1690172449236119?thread_ts=1690171425.054239&cid=C02NK2DQWUX

asked another c+ to review the PR

**Comment by mananjadhav** [comment]
Assigning myself.

**Comment by gadhiyamanan** [comment]
i think there is small [regression](https://expensify.slack.com/archives/C049HHMV9SM/p1690436737183979)  
cc: @mananjadhav @danieldoglas @yh-0218

**Comment by mananjadhav** [comment]
Yes! @yh-0218 Can we please fix this? I did see and I thought we added this message recently for all cases?

**Comment by sophiepintoraetz** [comment]
Confirmation of payouts:
@yh-0218: $250 (I don't think these are eligible for timeline bonuses)
@mananjadhav: $250 (no timeline bonus)

@yh-0218 - I'll release payment once the regression is fixed (I've send you an offer in Upwork!)

**Comment by mananjadhav** [comment]
@yh-0218 Can you add a new PR to fix the regression? It is a deploy blocker.

**Comment by mountiny** [comment]
We should decrease the bounty here as there was a regression from this PR

**Comment by danieldoglas** [comment]
I don't think we have a fix for this already? @yh-0218

**Comment by yh-0218** [comment]
Got it @danieldoglas Thanks for your reply.

**Comment by sophiepintoraetz** [comment]
@yh-0218 - we typically halve payments for introducing a regression - so the updated amounts are:
- @mananjadhav = $125 - via NewDot request
- @yh-0218 = $125 via upwork.

**Comment by sophiepintoraetz** [comment]
Ah, just realised we're subject to a regression period (@yh-0218 - you've been paid $125 already - let's hope another regression doesn't come through!)

**Comment by mananjadhav** [comment]
I've raised a request on NewDot for myself.

**Comment by sophiepintoraetz** [comment]
payments done.

**Comment by JmillsExpensify** [comment]
Reviewed the details for @mananjadhav. This is approved for payment via NewDot.
